drivers/adc/adc_ad559x.c

```diff
@@ -275,7 +275,7 @@ static int adc_ad559x_init(const struct device *dev)
 	return 0;
 }
 
-static const struct adc_driver_api adc_ad559x_api = {
+static DEVICE_API(adc_driver_api, adc_ad559x_api) = {
```

Comment:
I insist that we should be hiding implementation details, ie, the `_driver_api` suffix.

Reply:
to enforce consistency in the structure naming? I also think it's a good idea

Reply:
Yes, also because it’s a requirement, so it’s kind of wrong to pass the responsibility to the users to name API structs correctly.

Reply:
I did not catch that PR, sorry. I'll modify with the `_driver_api` suffix, and let the python script throw an error otherwise then.

Reply:
Updated the PR to keep `_driver_api` suffix requirements.